nifi-nar-bundles/nifi-marklogic-bundle/nifi-marklogic-processors/src/main/java/org/apache/nifi/marklogic/processor/PutMarkLogicRecord.java

```diff
@@ -207,35 +208,11 @@ protected WriteEvent buildWriteEvent(
             final FlowFile flowFile,
             String uri,
             final BytesHandle contentHandle,
-            final Map<String, String> additionalAttributes
+            final DocumentMetadataHandle metadata
     ) {
-        final String prefix = context.getProperty(URI_PREFIX).evaluateAttributeExpressions(flowFile).getValue();
```

Comment:
This was mostly duplicated code - moved it to protected methods in PutMarkLogic for reuse.